web/annotation_layer_builder.js

```diff
@@ -119,6 +120,61 @@ class AnnotationLayerBuilder {
       return;
     }
 
+    const uniqueLinks = inferredLinkAnnotations.filter(link => {
```

Comment:
Given that the viewer components must still work, this code cannot assume that `inferredLinkAnnotations` is provided.

Also, having all of this code in the middle of `AnnotationLayerBuilder.prototype.render` seems unfortunate and I'd suggest placing the new helper code in a *private method* in this class.
That'd allow this to be written as follows instead:
```suggestion
    if (inferredLinkAnnotations?.length) {
      annotations.push(...this.#parseInferredLinks(inferredLinkAnnotations);
    }
```